### DISCUSSION TITLE
QSWATMOD2 2.7 released

### DISCUSSION THREAD

**Discussion by spark-hydro** [spark-brc/QSWATMOD2#16] [Announcements]
An updated version of QSWATMOD2 has been released and available at https://github.com/spark-brc/QSWATMOD2-plugin/releases/download/v2.7.2/QSWATMOD2.exe
QSWATMOD now uses a CSV format for measurement files containing stream discharge and water levels (groundwater levels or depth to water). QSWATMOD will automatically detect files with the "*.obd.csv" extension in your working directory, allowing users to select and match them with simulated output. The recommended names for the parameter estimation process are as follows:

- Observed stream discharge:
  - daily: stf_day.obd.csv
  - monthly average: stf_mon.obd.csv (use the end date of the month) 
  - annual average: stf_ann.obd.csv (use the end date of the year) 
- Observed groundwater level:
  - daily: gwl_day.obd.csv
  - monthly average: gwl_mon.obd.csv (use the end date of the month) 
  - annual average: gwl_ann.obd.csv (use the end date of the year) 
- Observed depth to water (values must be negative: groundwater level - land surface elevation):
  - daily: dtw_day.obd.csv
  - monthly average: dtw_mon.obd.csv (use the end date of the month) 
  - annual average: dtw_ann.obd.csv (use the end date of the year) 

Additionally, it's worth noting that missing values in the CSV files can be represented by either "-999" or an empty field ("").

Here is an example format of the measurement files
<img src="https://github.com/spark-brc/QSWATMOD2-plugin/assets/17608997/1a24df20-83ba-436a-a618-5b77b3003530" width="500">

and a workflow.
![ezgif com-optimize](https://github.com/spark-brc/QSWATMOD2-plugin/assets/17608997/d15831e9-6292-4add-bc5d-b6e51d5bfcfc)

**Comment by ClementNdome** [comment]
I am trying to prepare my **modflow.obd,** unfortunately it is not reflecting on checking the box for prepared measurement files in the hydrograph for hydraulic head.  I am running on the latest QSWATMOD2 2.7 version.
![{362DA6D6-4AC4-4878-8CFE-AA9EFE963DD6}](https://github.com/user-attachments/assets/f91de2bd-e0b1-41c6-a70e-8f862738f371)
Another concern is the exporting the hydraulic head/ recharge (m3/day); the duration part is not active on my side thus cant cant recharge of different years/months/days. 
![{6FFBCA0C-5C7D-4412-B7FC-310A7B9859DA}](https://github.com/user-attachments/assets/81691ca6-0f84-4a32-ba66-5c901b1b8283)
Kindly help

**Reply by spark-hydro** [reply]
The current version reads the CSV file format. When your model folder (SWAT-MODFLOW) includes the file with the extension (*.obd.csv), then QSWATMOD will show the list of the files. Please check the sample files at the following link: https://github.com/spark-brc/QSWATMOD2-plugin/discussions/14.
